frontend/src/utils/fetchMajorsAndPlans.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  fetchPlansPendingAction,
+  fetchPlansSuccessAction,
+  fetchPlansErrorAction,
+} from "../state/actions/plansActions";
+import {
+  fetchMajorsPendingAction,
+  fetchMajorsSuccessAction,
+  fetchMajorsErrorAction,
+} from "../state/actions/majorsActions";
+import { Dispatch } from "redux";
+import { Major } from "../models/types";
+import { majorIds, majorMap } from "../majors";
+import { Schedule } from "../models/types";
+
+const majorSchema: string[] = majorIds.map((majorId: string) => {
+  return `major(majorId: "${majorId}") {
+        latestOccurrence {
+            requirements
+            plansOfStudy
+        }
+    }`;
+});
```

Comment:
This should be done using a GraphQL client like Relay or Apollo, should be done in a separate issue.

Reply:
I will make a card for this on our trello board